spring-cloud-sleuth-zipkin-stream/src/main/java/org/springframework/cloud/sleuth/zipkin/stream/SamplingZipkinSpanIterator.java

```diff
@@ -138,4 +158,45 @@ public void remove() {
 		}
 		return zipkinSpan.build();
 	}
+
+	private static void addLocalComponentAnnotation(Span span, Builder zipkinSpan,
+			Endpoint ep) {
+		String processId = span.getProcessId() != null
+				? span.getProcessId().toLowerCase()
+				: ZipkinMessageListener.UNKNOWN_PROCESS_ID;
+		zipkinSpan.addBinaryAnnotation(
+				BinaryAnnotation.create(Constants.LOCAL_COMPONENT, processId, ep));
+	}
+
+	private static void addServerAddressAnnotation(zipkin.Span.Builder zipkinSpan,
+			Endpoint ep) {
+		BinaryAnnotation component = new BinaryAnnotation.Builder()
```

Comment:
Hmm but this `BinaryAnnotation.address` allows only to take info from `Endpoint`. If one explicitly provides the service name from `peer.service` it will be ignored